pkg/tfbridge/provider.go

```diff
@@ -1665,3 +1660,12 @@ func transformFromState(
 	}
 	return o, nil
 }
+
+// If a custom timeout has been set for this method, overwrite the default timeout.
+func newTimeoutOverrides(key shim.TimeoutKey, maybeTimeoutSeconds float64) map[shim.TimeoutKey]time.Duration {
+	timeoutOverrides := map[shim.TimeoutKey]time.Duration{}
+	if maybeTimeoutSeconds != 0 {
+		timeoutOverrides[shim.TimeoutCreate] = time.Duration(maybeTimeoutSeconds * float64(time.Second))
+	}
```

Comment:
This reads a lot clearer to me.
```suggestion
	// We interpret 0 to mean no timeout set.
	if maybeTimeoutSeconds != 0 {
		timeoutSeconds := time.Duration(maybeTimeoutSeconds)
		timeoutOverrides[shim.TimeoutCreate] = timeoutSeconds * time.Second
	}
```

Reply:
Eh no, float64->int64 cast in time.Duration(float64) will lose precision. I took the above formula from a SO answer..